py-greenlet@0.4.13 build fails on power arch: platform/switch_ppc64_linux.h

Building `py-greenlet@0.14.3` fails using `spack@develop` (c5b4a667d3e8d44efe5395c33a5c57ec66dc5056 from `Thu Dec 17 16:56:39 2020 -0500`).

I believe this is due to a fairly recent change (post ~ 2020-12-14)

System info:
* `Ubuntu 18.04` for `ppc64le`
* `GCC 7.5.0`
* Docker container image: `ecpe4s/ubuntu18.04-runner-ppc64le:2020-12-01`

Concrete spec: [py-greenlet-5gtlvi.yml.txt](https://github.com/spack/spack/files/5712651/py-greenlet-5gtlvi.yml.txt)
Build log: [py-greenlet-build-out.txt](https://github.com/spack/spack/files/5712667/py-greenlet-build-out.txt)

```
$> spack install -f ./py-greenlet-5gtlvi.yml
...
==> Installing py-greenlet-0.4.13-6qd3bhzb723ij5pzq5pj46fnowefblhq
==> Fetching https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/_source-cache/archive/0f/0fef83d43bf87a5196c91e73cb9772f945a4caaff91242766c5916d1dd1381e4.tar.gz
############################################################################################################################################################ 100.0%
==> py-greenlet: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 1:
    '/opt/spack/opt/spack/linux-ubuntu18.04-ppc64le/gcc-7.5.0/python-3.8.6-ywxntapm3srwjfo4yd5s3gdu76d3w3nm/bin/python3.8' '-s' 'setup.py' '--no-user-cfg' 'build'

3 errors found in build log:
     6     creating build
     7     creating build/temp.linux-ppc64le-3.8
     8     /opt/spack/lib/spack/env/gcc/gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -fPIC -I/opt/spack/opt/spack/linux-ubuntu18.04-p
           pc64le/gcc-7.5.0/python-3.8.6-ywxntapm3srwjfo4yd5s3gdu76d3w3nm/include/python3.8 -c greenlet.c -o build/temp.linux-ppc64le-3.8/greenlet.o -fno-tree-dom
           inator-opts
     9     In file included from slp_platformselect.h:16:0,
     10                     from greenlet.c:328:
     11    platform/switch_ppc64_linux.h: In function 'slp_switch':
  >> 12    platform/switch_ppc64_linux.h:72:5: error: PIC register clobbered by 'r2' in 'asm'
     13         __asm__ volatile ("" : : : REGS_TO_SAVE);
     14         ^~~~~~~
  >> 15    platform/switch_ppc64_linux.h:85:5: error: PIC register clobbered by 'r2' in 'asm'
     16         __asm__ volatile ("" : : : REGS_TO_SAVE);
     17         ^~~~~~~
  >> 18    error: command '/opt/spack/lib/spack/env/gcc/gcc' failed with exit status 1

See build log for details:
  /tmp/root/spack-stage/spack-stage-py-greenlet-0.4.13-6qd3bhzb723ij5pzq5pj46fnowefblhq/spack-build-out.txt
```

@adamjstewart @becker33

Any ideas?

--- a/var/spack/repos/builtin/packages/py-greenlet/package.py
+++ b/var/spack/repos/builtin/packages/py-greenlet/package.py
@@ -10,6 +10,7 @@ class PyGreenlet(PythonPackage):
     """Lightweight in-process concurrent programming"""
 
     homepage = "https://github.com/python-greenlet/greenlet"
-    url      = "https://pypi.io/packages/source/g/greenlet/greenlet-0.4.13.tar.gz"
+    url      = "https://pypi.io/packages/source/g/greenlet/greenlet-0.4.17.tar.gz"
 
+    version('0.4.17', sha256='41d8835c69a78de718e466dd0e6bfd4b46125f21a67c3ff6d76d8d8059868d6b')
     version('0.4.13', sha256='0fef83d43bf87a5196c91e73cb9772f945a4caaff91242766c5916d1dd1381e4')

--- a/var/spack/repos/builtin/packages/py-greenlet/package.py
+++ b/var/spack/repos/builtin/packages/py-greenlet/package.py
@@ -14,3 +14,5 @@ class PyGreenlet(PythonPackage):
 
     version('0.4.17', sha256='41d8835c69a78de718e466dd0e6bfd4b46125f21a67c3ff6d76d8d8059868d6b')
     version('0.4.13', sha256='0fef83d43bf87a5196c91e73cb9772f945a4caaff91242766c5916d1dd1381e4')
+
+    depends_on('python', type=('build', 'link', 'run'))

